custom_model_runner/datarobot_drum/drum/data_marshalling.py

```diff
@@ -0,0 +1,164 @@
+                return e_l
+
+    return [get_corresponding_expected_label(_l) for _l in actual_labels]
+
+
+def _standardize(label):
+    # First, hope it's a float
+    try:
+        return float(label)
+    except ValueError:
+        pass
+
+    # Maybe if its a boolean we can make it floaty anyways
+    try:
+        return float(strtobool(label))
+    except ValueError:
+        pass
+
+    # Okay lets just do a str.lower
+    assert isinstance(label, str)
+    return label.lower()
+
+
+def _classification_marshal_preds(predictions, request_labels, model_labels):
+    request_labels = _marshal_labels(request_labels, model_labels)
+    predictions = _duplicate_1d_probabilities(predictions, request_labels)
+    _validate_amount_of_columns(request_labels, predictions)
+    _validate_probabilities_make_sense(predictions)
+    return pd.DataFrame(predictions, columns=request_labels)
+
+
+def _single_col_marshal_preds(predictions):
+    _validate_predictions_are_one_dimensional(predictions)
+    return pd.DataFrame(predictions, columns=[REGRESSION_PRED_COLUMN])
+
+
+def _validate_dimensionality_and_type(predictions):
+    if not isinstance(predictions, np.ndarray):
+        raise DrumCommonException(
+            f"predictions must return a np array, but received {type(predictions)}"
+        )
+    if len(predictions.shape) == 1:
+        predictions = predictions.reshape(-1, 1)
+    if len(predictions.shape) != 2:
+        raise DrumCommonException(
+            f"predictions must return a 2 dimensional array, but we received a {len(predictions.shape)} dimensional array"
+        )
+    return predictions
+
+
+def _validate_predictions_are_one_dimensional(predictions):
+    if predictions.shape[1] != 1:
+        raise DrumCommonException(
+            f"Regression and anomaly predictions must contain only 1 column. Your predictions have {predictions.shape[1]} columns"
+        )
+
+
+def _validate_probabilities_make_sense(predictions):
+    if (predictions < 0).any():
+        raise DrumCommonException("Your predictions have negative values")
+    if (predictions > 1).any():
+        raise DrumCommonException("Your predictions have values greater than 1")
+    added_probs = predictions.sum(axis=1)
+    good_preds = np.isclose(added_probs, 1)
+    if not np.all(good_preds):
+        bad_rows = predictions[~good_preds]
+        raise DrumCommonException(
+            "Your prediction probabilities do not add up to 1. \n{}".format(bad_rows)
+        )
+
+
+def _validate_amount_of_columns(labels_to_use, predictions):
+    if predictions.shape[1] != len(labels_to_use):
+        raise DrumCommonException(
+            "Your predictions must return the "
+            "probability distribution for the correct number of class labels. "
+            "Expected {} columns, but received {}".format(len(labels_to_use), predictions.shape[1])
+        )
+
+
+def _duplicate_1d_probabilities(predictions, labels):
+    if predictions.shape[1] == 1:
```

Comment:
we should make sure target type is binary here

Reply:
we actually can duplicate for the multiclass case too where there are only 2 labels.